---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r setup, include = FALSE}
knitr::opts_chunk$set(
    comment = "#>",
    collapse = TRUE,
    echo = TRUE,
    cache = TRUE,
    cache.lazy = FALSE
)
```

# Tidy Project Achilles Data (tidy_Achilles)

**author: Joshua H. Cook**  
**last updated: June 12, 2019**

**[GitHub repository](https://github.com/jhrcook/tidy_Achilles)**

This is "tidy" data from the Broad's [Project Achilles](https://depmap.org/portal/achilles/). The data files are in the "data" directory and exaplined below.

To download this repository, run the following command on the command line.

```bash
git clone https://github.com/jhrcook/tidy_Achilles.git
```

## Data Source

All data was downloaded from the [Broad's DepMap data repository](https://depmap.org/portal/download/all/).\
**web address:** https://depmap.org/portal/download/all/

You can also query genes, cell lines, lineages, etc. from their [website](https://depmap.org/portal/)\
**web address:** https://depmap.org/portal/

The cell line information was obtained from the [Cancer Cell Line Encyclopedia](https://portals.broadinstitute.org/ccle) - their query portal is really useful, too. More data on the cell lines can be downloaded from their website. If you have an requests for data to add to this repo, please open a GitHub [issue](https://github.com/jhrcook/tidy_Achilles/issues).\
**web address:** https://portals.broadinstitute.org/ccle

---

## Raw Data

### Cell Line Data

The following files contain information about the cell lines in the CCLE. Some, but not all, have been screen by the DepMap Project.

* sample\_info.csv - information on the cell lines
* CCLE\_mutations.csv.tar.gz - mutations of the cell lines
* CCLE\_expression.csv.tar.gz - gene expression of the cell lines (in TPM)
* CCLE\_gene\_cn.csv.tar.gz - copy number of each gene in the cell lines


### Achilles Genetic Dependencies

The Achilles project used RNAi to deplete each gene in the cell lines, but has since transitioned to a CRISPR loss-of-function screen. That data is available in this repository. The following files hold information on the effect of knocking-out each gene on the survival of the cell line.

* Achilles\_gene\_dependency.csv.tar.gz - probability that knocking out the gene has a real depletion effect
* Achilles\_gene\_effect.csv.tar.gz - CERES data with principle components strongly related to known batch effects removed, then shifted and scaled per cell line so the median nonessential KO effect is 0 and the median essential KO effect is -1


### Achilles Guides

The Achilles project used RNAi to deplete each gene in the cell lines, but has since transitioned to a CRISPR loss-of-function screen. That data is available in this repository. The following files hold information on the guides used in the CRISPR screen.

* Achilles\_guide\_efficacy.csv - CERES inferred efficacy for the guide
* Achilles\_guide\_map.csv - where the guide targets on the genome (including gene name)
* Achilles\_dropped\_guides.csv - guides that were dropped, usually for likely off-target effects


### Achilles misc.

* Achilles\_replicate\_map.csv - indicates which processing batch the replicate belongs to and therefore which pDNA reference it should be compared with


### Essential Genes

The data collected from the Achilles project was used to determine which genes were broadly essential.

* common\_essentials.csv - genes used as positive controls, intersection of Biomen (2014) and Hart (2015) essentials; the scores of these genes are used as the dependent distribution for inferring dependency probability
* Achilles\_common\_essentials.csv - genes identified as dependencies in all lines
* nonessentials.csv - genes used as negative controls (Hart (2014) nonessentials)


### To be added

Any others? Open an [issue](https://github.com/jhrcook/tidy_Achilles/issues) for any requested data to include.

---

## Tidy Data

```{r load_libs, message = FALSE, warning = FALSE}
library(tidyverse)
```

All processing was done in "data_preparation.R". The tidy data were stored as "tibbles" (`tbl_df` instead of R's standard `data.frame` object) in RDS files. They can all be read directly into R.

```{r example_readRDS, eval = FALSE}
library(tibble)
readRDS("data/example_data_table.tib")
```

More information in the "tidy data" format can be found in [*R for Data Science - Tidy data*](https://r4ds.had.co.nz/tidy-data.html). 

### Cell Line Data

#### Meta data

The following data frame holds meta data on the cell lines used in the CRISPR screen (an other cell lines in the CCLE). Use the `dep_map_id` column for unique identification and matching with the data frames included in this repository. The columns `disease` and `disease_sutype` provide information on where the cell line originated.

```{r show_metadata}
readRDS(file.path("data", "cell_line_metadata.tib"))
```

#### Mutations

All of the cell lines in the CCLE have been whole exome sequenced. Their mutations are contained in the following file. Both the Hugo and Entrez gene identifiers are available (`hugo_symbol` and `entrez_gene_id`, respectively). Use the `dep_map_id` column for unique identification and matching with the data frames included in this repository. The specific mutations to the proteins are in the `protein_change` column.

```{r show_celllinemutations}
readRDS(file.path("data", "cell_line_mutations.tib"))
```

#### Copy Number

Due to size restrictions, the gene copy number data was separated into multiple files by tissue of origin of the cell line. The Hugo gene identifier is available in the `gene` column. Use the `dep_map_id` column for unique identification and matching with the data frames included in this repository. The copy number information is in the `copy_number` column.

```{r show_copynumber}
readRDS(file.path("data", "copy_number", "PANCREAS_copynum.tib"))
```

#### Gene Expression

Due to size restrictions, the gene expression (in TPM) data was separated into multiple files by tissue of origin of the cell line. The Hugo gene identifier is available in the `gene` column. Use the `dep_map_id` column for unique identification and matching with the data frames included in this repository. The expression values are in the `gene_expression` column.

```{r show_geneexpr}
readRDS(file.path("data", "gene_expression", "PANCREAS_geneexpr.tib"))
```


### Achilles

A cell line is dependent on a gene if the deletion of the gene causes a decrease in vitality of the cell line. The Achilles project is using a genome-wide CRISPR-Cas9 loss-of-function screen to test hundreds of cell line's dependencies.

The `Achilles_gene_dependency.tib` contains the probability that knocking out the gene has a real depletion effect. The Hugo gene identifier is available in the `gene` column. Use the `dep_map_id` column for unique identification and matching with the data frames included in this repository. The dependency score is in the `gene_dependency` column. This file includes the CCLE sample information, too.

```{r show_achillesdep}
readRDS(file.path("data", "Achilles_gene_dependency.tib"))
```

```{r hist_achillesdep, echo = FALSE, warning = FALSE, message = FALSE}
dat <- readRDS(file.path("data", "Achilles_gene_dependency.tib"))
dat %>%
    mutate(tissue = str_to_lower(str_replace_all(tissue, "_", " "))) %>%
    ggplot(aes(x = tissue, y = -log10(gene_dependency))) +
    geom_boxplot(aes(fill = tissue),
                 outlier.size = 0.7, outlier.color = "grey40") +
    theme_minimal() +
    theme(
        axis.text.x = element_text(angle = 60, size = 6, hjust = 1)
    ) +
    scale_fill_hue(guide = FALSE)
```

The depletion effect of targeting each gene is in `Achilles_gene_effect.tib`. This contains CERES data with principle components strongly related to known batch effects removed, then shifted and scaled per cell line so the median nonessential knock-out effect is 0 and the median essential knock-out effect is -1. The gene effect values are in the `gene_effect` column.

```{r show_achilleseffect}
readRDS(file.path("data", "Achilles_gene_effect.tib"))
```

```{r hist_achilleseffect, echo = FALSE, warning = FALSE, message = FALSE}
dat <- readRDS(file.path("data", "Achilles_gene_effect.tib"))
dat %>%
    mutate(tissue = str_to_lower(str_replace_all(tissue, "_", " "))) %>%
    ggplot(aes(x = tissue, y = gene_effect)) +
    geom_violin(aes(fill = tissue)) +
    geom_hline(yintercept = 0, size = 0.8, color = "grey25", linetype = 2) +
    theme_minimal() +
    theme(
        axis.text.x = element_text(angle = 60, size = 6, hjust = 1)
    ) +
    scale_fill_hue(guide = FALSE)
```

The DepMap released information on the guides they used including, mapping information, efficacy, and those that they removed (for various reasons, though likely for off-target effects). These data were collated into one data frame.

```{r show_guidedata}
readRDS(file.path("data", "Achilles_guides.tib"))
```

The DepMap project released a file containing which batch each sample (cell line) was processed in. Note, the scores in `Achilles_gene_effect.tib` have already been corrected for batch effects.

```{r show_batches}
readRDS(file.path("data", "Achilles_replicate_map.tib"))
```

Finally, the DepMap created three lists of genes according tho their essentialiaty. Those were merged into one data frame, here. The `common_essential` column indicates which genes were previously known to be essential in all cells, while the `achilles_essential` column indicates which genes were found to be pan-essential during the Achilles screen. The `nonessential` column indicates which genes are universally non-essential

```{r show_essentials}
readRDS(file.path("data", "gene_essentiality.tib"))
```

The following Venn diagram shows the overlap of these three groups.

```{r venn_essentials, echo = FALSE, warning = FALSE, message = FALSE}
library(VennDiagram)
dat <- readRDS(file.path("data", "gene_essentiality.tib"))
known_essentials <- filter(dat, common_essential) %>% pull(gene) %>% unlist() %>% unique()
achilles_essentials <- filter(dat, achilles_essential) %>% pull(gene) %>% unlist() %>% unique()
nonessentials <- filter(dat, nonessential) %>% pull(gene) %>% unlist() %>% unique()
venn_plot <- venn.diagram(
    x = list(known_essentials, achilles_essentials, nonessentials),
    category.names = c(
        "known essentials", "Achilles essentials", "non-essentials"
    ),
    filename = NULL,
    fill = c('dodgerblue', 'tomato', 'mediumseagreen')
)
grid.newpage()
grid.draw(venn_plot)
```

---

If there are any mistake or places for explanation, please open an [issue](https://github.com/jhrcook/tidy_Achilles/issues) or create a pull request if you want to address it yourself.
