<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta name="viewport" content="width=device-width, initial-scale=1">

<style type="text/css">
@font-face {
font-family: octicons-link;
src: url(data:font/woff;charset=utf-8;base64,d09GRgABAAAAAAZwABAAAAAACFQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABEU0lHAAAGaAAAAAgAAAAIAAAAAUdTVUIAAAZcAAAACgAAAAoAAQAAT1MvMgAAAyQAAABJAAAAYFYEU3RjbWFwAAADcAAAAEUAAACAAJThvmN2dCAAAATkAAAABAAAAAQAAAAAZnBnbQAAA7gAAACyAAABCUM+8IhnYXNwAAAGTAAAABAAAAAQABoAI2dseWYAAAFsAAABPAAAAZwcEq9taGVhZAAAAsgAAAA0AAAANgh4a91oaGVhAAADCAAAABoAAAAkCA8DRGhtdHgAAAL8AAAADAAAAAwGAACfbG9jYQAAAsAAAAAIAAAACABiATBtYXhwAAACqAAAABgAAAAgAA8ASm5hbWUAAAToAAABQgAAAlXu73sOcG9zdAAABiwAAAAeAAAAME3QpOBwcmVwAAAEbAAAAHYAAAB/aFGpk3jaTY6xa8JAGMW/O62BDi0tJLYQincXEypYIiGJjSgHniQ6umTsUEyLm5BV6NDBP8Tpts6F0v+k/0an2i+itHDw3v2+9+DBKTzsJNnWJNTgHEy4BgG3EMI9DCEDOGEXzDADU5hBKMIgNPZqoD3SilVaXZCER3/I7AtxEJLtzzuZfI+VVkprxTlXShWKb3TBecG11rwoNlmmn1P2WYcJczl32etSpKnziC7lQyWe1smVPy/Lt7Kc+0vWY/gAgIIEqAN9we0pwKXreiMasxvabDQMM4riO+qxM2ogwDGOZTXxwxDiycQIcoYFBLj5K3EIaSctAq2kTYiw+ymhce7vwM9jSqO8JyVd5RH9gyTt2+J/yUmYlIR0s04n6+7Vm1ozezUeLEaUjhaDSuXHwVRgvLJn1tQ7xiuVv/ocTRF42mNgZGBgYGbwZOBiAAFGJBIMAAizAFoAAABiAGIAznjaY2BkYGAA4in8zwXi+W2+MjCzMIDApSwvXzC97Z4Ig8N/BxYGZgcgl52BCSQKAA3jCV8CAABfAAAAAAQAAEB42mNgZGBg4f3vACQZQABIMjKgAmYAKEgBXgAAeNpjYGY6wTiBgZWBg2kmUxoDA4MPhGZMYzBi1AHygVLYQUCaawqDA4PChxhmh/8ODDEsvAwHgMKMIDnGL0x7gJQCAwMAJd4MFwAAAHjaY2BgYGaA4DAGRgYQkAHyGMF8NgYrIM3JIAGVYYDT+AEjAwuDFpBmA9KMDEwMCh9i/v8H8sH0/4dQc1iAmAkALaUKLgAAAHjaTY9LDsIgEIbtgqHUPpDi3gPoBVyRTmTddOmqTXThEXqrob2gQ1FjwpDvfwCBdmdXC5AVKFu3e5MfNFJ29KTQT48Ob9/lqYwOGZxeUelN2U2R6+cArgtCJpauW7UQBqnFkUsjAY/kOU1cP+DAgvxwn1chZDwUbd6CFimGXwzwF6tPbFIcjEl+vvmM/byA48e6tWrKArm4ZJlCbdsrxksL1AwWn/yBSJKpYbq8AXaaTb8AAHja28jAwOC00ZrBeQNDQOWO//sdBBgYGRiYWYAEELEwMTE4uzo5Zzo5b2BxdnFOcALxNjA6b2ByTswC8jYwg0VlNuoCTWAMqNzMzsoK1rEhNqByEyerg5PMJlYuVueETKcd/89uBpnpvIEVomeHLoMsAAe1Id4AAAAAAAB42oWQT07CQBTGv0JBhagk7HQzKxca2sJCE1hDt4QF+9JOS0nbaaYDCQfwCJ7Au3AHj+LO13FMmm6cl7785vven0kBjHCBhfpYuNa5Ph1c0e2Xu3jEvWG7UdPDLZ4N92nOm+EBXuAbHmIMSRMs+4aUEd4Nd3CHD8NdvOLTsA2GL8M9PODbcL+hD7C1xoaHeLJSEao0FEW14ckxC+TU8TxvsY6X0eLPmRhry2WVioLpkrbp84LLQPGI7c6sOiUzpWIWS5GzlSgUzzLBSikOPFTOXqly7rqx0Z1Q5BAIoZBSFihQYQOOBEdkCOgXTOHA07HAGjGWiIjaPZNW13/+lm6S9FT7rLHFJ6fQbkATOG1j2OFMucKJJsxIVfQORl+9Jyda6Sl1dUYhSCm1dyClfoeDve4qMYdLEbfqHf3O/AdDumsjAAB42mNgYoAAZQYjBmyAGYQZmdhL8zLdDEydARfoAqIAAAABAAMABwAKABMAB///AA8AAQAAAAAAAAAAAAAAAAABAAAAAA==) format('woff');
}
body {
-webkit-text-size-adjust: 100%;
text-size-adjust: 100%;
color: #333;
font-family: "Helvetica Neue", Helvetica, "Segoe UI", Arial, freesans, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
font-size: 16px;
line-height: 1.6;
word-wrap: break-word;
}
a {
background-color: transparent;
}
a:active,
a:hover {
outline: 0;
}
strong {
font-weight: bold;
}
h1 {
font-size: 2em;
margin: 0.67em 0;
}
img {
border: 0;
}
hr {
box-sizing: content-box;
height: 0;
}
pre {
overflow: auto;
}
code,
kbd,
pre {
font-family: monospace, monospace;
font-size: 1em;
}
input {
color: inherit;
font: inherit;
margin: 0;
}
html input[disabled] {
cursor: default;
}
input {
line-height: normal;
}
input[type="checkbox"] {
box-sizing: border-box;
padding: 0;
}
table {
border-collapse: collapse;
border-spacing: 0;
}
td,
th {
padding: 0;
}
* {
box-sizing: border-box;
}
input {
font: 13px / 1.4 Helvetica, arial, nimbussansl, liberationsans, freesans, clean, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
}
a {
color: #4078c0;
text-decoration: none;
}
a:hover,
a:active {
text-decoration: underline;
}
hr {
height: 0;
margin: 15px 0;
overflow: hidden;
background: transparent;
border: 0;
border-bottom: 1px solid #ddd;
}
hr:before {
display: table;
content: "";
}
hr:after {
display: table;
clear: both;
content: "";
}
h1,
h2,
h3,
h4,
h5,
h6 {
margin-top: 15px;
margin-bottom: 15px;
line-height: 1.1;
}
h1 {
font-size: 30px;
}
h2 {
font-size: 21px;
}
h3 {
font-size: 16px;
}
h4 {
font-size: 14px;
}
h5 {
font-size: 12px;
}
h6 {
font-size: 11px;
}
blockquote {
margin: 0;
}
ul,
ol {
padding: 0;
margin-top: 0;
margin-bottom: 0;
}
ol ol,
ul ol {
list-style-type: lower-roman;
}
ul ul ol,
ul ol ol,
ol ul ol,
ol ol ol {
list-style-type: lower-alpha;
}
dd {
margin-left: 0;
}
code {
font-family: Consolas, "Liberation Mono", Menlo, Courier, monospace;
font-size: 12px;
}
pre {
margin-top: 0;
margin-bottom: 0;
font: 12px Consolas, "Liberation Mono", Menlo, Courier, monospace;
}
.select::-ms-expand {
opacity: 0;
}
.octicon {
font: normal normal normal 16px/1 octicons-link;
display: inline-block;
text-decoration: none;
text-rendering: auto;
-webkit-font-smoothing: antialiased;
-moz-osx-font-smoothing: grayscale;
-webkit-user-select: none;
-moz-user-select: none;
-ms-user-select: none;
user-select: none;
}
.octicon-link:before {
content: '\f05c';
}
.markdown-body:before {
display: table;
content: "";
}
.markdown-body:after {
display: table;
clear: both;
content: "";
}
.markdown-body>*:first-child {
margin-top: 0 !important;
}
.markdown-body>*:last-child {
margin-bottom: 0 !important;
}
a:not([href]) {
color: inherit;
text-decoration: none;
}
.anchor {
display: inline-block;
padding-right: 2px;
margin-left: -18px;
}
.anchor:focus {
outline: none;
}
h1,
h2,
h3,
h4,
h5,
h6 {
margin-top: 1em;
margin-bottom: 16px;
font-weight: bold;
line-height: 1.4;
}
h1 .octicon-link,
h2 .octicon-link,
h3 .octicon-link,
h4 .octicon-link,
h5 .octicon-link,
h6 .octicon-link {
color: #000;
vertical-align: middle;
visibility: hidden;
}
h1:hover .anchor,
h2:hover .anchor,
h3:hover .anchor,
h4:hover .anchor,
h5:hover .anchor,
h6:hover .anchor {
text-decoration: none;
}
h1:hover .anchor .octicon-link,
h2:hover .anchor .octicon-link,
h3:hover .anchor .octicon-link,
h4:hover .anchor .octicon-link,
h5:hover .anchor .octicon-link,
h6:hover .anchor .octicon-link {
visibility: visible;
}
h1 {
padding-bottom: 0.3em;
font-size: 2.25em;
line-height: 1.2;
border-bottom: 1px solid #eee;
}
h1 .anchor {
line-height: 1;
}
h2 {
padding-bottom: 0.3em;
font-size: 1.75em;
line-height: 1.225;
border-bottom: 1px solid #eee;
}
h2 .anchor {
line-height: 1;
}
h3 {
font-size: 1.5em;
line-height: 1.43;
}
h3 .anchor {
line-height: 1.2;
}
h4 {
font-size: 1.25em;
}
h4 .anchor {
line-height: 1.2;
}
h5 {
font-size: 1em;
}
h5 .anchor {
line-height: 1.1;
}
h6 {
font-size: 1em;
color: #777;
}
h6 .anchor {
line-height: 1.1;
}
p,
blockquote,
ul,
ol,
dl,
table,
pre {
margin-top: 0;
margin-bottom: 16px;
}
hr {
height: 4px;
padding: 0;
margin: 16px 0;
background-color: #e7e7e7;
border: 0 none;
}
ul,
ol {
padding-left: 2em;
}
ul ul,
ul ol,
ol ol,
ol ul {
margin-top: 0;
margin-bottom: 0;
}
li>p {
margin-top: 16px;
}
dl {
padding: 0;
}
dl dt {
padding: 0;
margin-top: 16px;
font-size: 1em;
font-style: italic;
font-weight: bold;
}
dl dd {
padding: 0 16px;
margin-bottom: 16px;
}
blockquote {
padding: 0 15px;
color: #777;
border-left: 4px solid #ddd;
}
blockquote>:first-child {
margin-top: 0;
}
blockquote>:last-child {
margin-bottom: 0;
}
table {
display: block;
width: 100%;
overflow: auto;
word-break: normal;
word-break: keep-all;
}
table th {
font-weight: bold;
}
table th,
table td {
padding: 6px 13px;
border: 1px solid #ddd;
}
table tr {
background-color: #fff;
border-top: 1px solid #ccc;
}
table tr:nth-child(2n) {
background-color: #f8f8f8;
}
img {
max-width: 100%;
box-sizing: content-box;
background-color: #fff;
}
code {
padding: 0;
padding-top: 0.2em;
padding-bottom: 0.2em;
margin: 0;
font-size: 85%;
background-color: rgba(0,0,0,0.04);
border-radius: 3px;
}
code:before,
code:after {
letter-spacing: -0.2em;
content: "\00a0";
}
pre>code {
padding: 0;
margin: 0;
font-size: 100%;
word-break: normal;
white-space: pre;
background: transparent;
border: 0;
}
.highlight {
margin-bottom: 16px;
}
.highlight pre,
pre {
padding: 16px;
overflow: auto;
font-size: 85%;
line-height: 1.45;
background-color: #f7f7f7;
border-radius: 3px;
}
.highlight pre {
margin-bottom: 0;
word-break: normal;
}
pre {
word-wrap: normal;
}
pre code {
display: inline;
max-width: initial;
padding: 0;
margin: 0;
overflow: initial;
line-height: inherit;
word-wrap: normal;
background-color: transparent;
border: 0;
}
pre code:before,
pre code:after {
content: normal;
}
kbd {
display: inline-block;
padding: 3px 5px;
font-size: 11px;
line-height: 10px;
color: #555;
vertical-align: middle;
background-color: #fcfcfc;
border: solid 1px #ccc;
border-bottom-color: #bbb;
border-radius: 3px;
box-shadow: inset 0 -1px 0 #bbb;
}
.pl-c {
color: #969896;
}
.pl-c1,
.pl-s .pl-v {
color: #0086b3;
}
.pl-e,
.pl-en {
color: #795da3;
}
.pl-s .pl-s1,
.pl-smi {
color: #333;
}
.pl-ent {
color: #63a35c;
}
.pl-k {
color: #a71d5d;
}
.pl-pds,
.pl-s,
.pl-s .pl-pse .pl-s1,
.pl-sr,
.pl-sr .pl-cce,
.pl-sr .pl-sra,
.pl-sr .pl-sre {
color: #183691;
}
.pl-v {
color: #ed6a43;
}
.pl-id {
color: #b52a1d;
}
.pl-ii {
background-color: #b52a1d;
color: #f8f8f8;
}
.pl-sr .pl-cce {
color: #63a35c;
font-weight: bold;
}
.pl-ml {
color: #693a17;
}
.pl-mh,
.pl-mh .pl-en,
.pl-ms {
color: #1d3e81;
font-weight: bold;
}
.pl-mq {
color: #008080;
}
.pl-mi {
color: #333;
font-style: italic;
}
.pl-mb {
color: #333;
font-weight: bold;
}
.pl-md {
background-color: #ffecec;
color: #bd2c00;
}
.pl-mi1 {
background-color: #eaffea;
color: #55a532;
}
.pl-mdr {
color: #795da3;
font-weight: bold;
}
.pl-mo {
color: #1d3e81;
}
kbd {
display: inline-block;
padding: 3px 5px;
font: 11px Consolas, "Liberation Mono", Menlo, Courier, monospace;
line-height: 10px;
color: #555;
vertical-align: middle;
background-color: #fcfcfc;
border: solid 1px #ccc;
border-bottom-color: #bbb;
border-radius: 3px;
box-shadow: inset 0 -1px 0 #bbb;
}
.task-list-item {
list-style-type: none;
}
.task-list-item+.task-list-item {
margin-top: 3px;
}
.task-list-item input {
margin: 0 0.35em 0.25em -1.6em;
vertical-align: middle;
}
:checked+.radio-label {
z-index: 1;
position: relative;
border-color: #4078c0;
}
.sourceLine {
display: inline-block;
}
code .kw { color: #000000; }
code .dt { color: #ed6a43; }
code .dv { color: #009999; }
code .bn { color: #009999; }
code .fl { color: #009999; }
code .ch { color: #009999; }
code .st { color: #183691; }
code .co { color: #969896; }
code .ot { color: #0086b3; }
code .al { color: #a61717; }
code .fu { color: #63a35c; }
code .er { color: #a61717; background-color: #e3d2d2; }
code .wa { color: #000000; }
code .cn { color: #008080; }
code .sc { color: #008080; }
code .vs { color: #183691; }
code .ss { color: #183691; }
code .im { color: #000000; }
code .va {color: #008080; }
code .cf { color: #000000; }
code .op { color: #000000; }
code .bu { color: #000000; }
code .ex { color: #000000; }
code .pp { color: #999999; }
code .at { color: #008080; }
code .do { color: #969896; }
code .an { color: #008080; }
code .cv { color: #008080; }
code .in { color: #008080; }
</style>
<style>
body {
  box-sizing: border-box;
  min-width: 200px;
  max-width: 980px;
  margin: 0 auto;
  padding: 45px;
  padding-top: 0px;
}
</style>

</head>

<body>

<!-- README.md is generated from README.Rmd. Please edit that file -->

<h1 id="tidy-project-achilles-data">Tidy Project Achilles Data</h1>
<p><strong>author: Joshua H. Cook</strong></p>
<p><strong>last updated: March 15, 2019</strong></p>
<p>This is “tidy” data from the Broad’s <a href="https://depmap.org/portal/achilles/">Project Achilles</a> with a focus of the various <em>RAS</em> alleles. The data files are in the “data” directory and exaplined below.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">library</span>(tidyverse)</a></code></pre></div>
<p>To download this repository, run the following command on the command line.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="fu">git</span> clone https://github.com/jhrcook/tidy_Achilles.git</a></code></pre></div>
<hr />
<h2 id="raw-data">Raw Data</h2>
<h3 id="depmap-2019q1-celllines_v2csv">DepMap-2019q1-celllines_v2.csv</h3>
<p>Information on cell lines from the Broad’s <a href="https://portals.broadinstitute.org/ccle">Cancer Cell Line Encyclopedia (CCLE)</a>. The tidy data is available in “cell_line_metadata.tib”.</p>
<h3 id="depmap_19q1_mutation_callscsv">depmap_19Q1_mutation_calls.csv</h3>
<p>Mutation data on the cell lines from the Broad’s <a href="https://portals.broadinstitute.org/ccle">Cancer Cell Line Encyclopedia (CCLE)</a>. The tidy data is available in “cell_line_mutations.tib”.</p>
<p>To access the file, decompress it with a GUI tools (usually double-click on Mac works) or use the command line.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="fu">gunzip</span> depmap_19Q1_mutation_calls.csv.gz</a></code></pre></div>
<h3 id="d2_combined_gene_dep_scorescsv">D2_combined_gene_dep_scores.csv</h3>
<p>The “dependency scores” calculated by the Achilles Project. This file is organized by target name in the first column and the following columns are the scores for each cell line. The tidy data is available in “synthetic_lethal.tib”.</p>
<p>To access the file, decompress it with a GUI tools (usually double-click on Mac works) or use the command line.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="fu">gunzip</span> D2_combined_gene_dep_scores.csv.gz</a></code></pre></div>
<hr />
<h2 id="tidy-data-tables">Tidy Data Tables</h2>
<p>All processing was done in “data_preparation.R”. The tidy data were stored as “tibbles” (<code>tbl_df</code>, instead of R’s standard data.frame object) in RDS files. They can all be read directly into R.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw">library</span>(tibble)</a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="kw">readRDS</span>(<span class="st">&quot;data/example_data_table.tib&quot;</span>)</a></code></pre></div>
<p>More information in the “tidy data” format can be found in <a href="https://r4ds.had.co.nz/tidy-data.html"><em>R for Data Science - Tidy data</em></a>.</p>
<h3 id="cell_line_metadatatib">cell_line_metadata.tib</h3>
<p>The information for each cell line from the Broad’s <a href="https://portals.broadinstitute.org/ccle">Cancer Cell Line Encyclopedia (CCLE)</a>.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">cell_line_metadata &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">file.path</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;cell_line_metadata.tib&quot;</span>))</a>
<a class="sourceLine" id="cb6-2" data-line-number="2">cell_line_metadata</a>
<a class="sourceLine" id="cb6-3" data-line-number="3"><span class="co">#&gt; # A tibble: 1,676 x 9</span></a>
<a class="sourceLine" id="cb6-4" data-line-number="4"><span class="co">#&gt;    DepMap_ID CCLE_Name Aliases COSMIC_ID Sanger_ID Primary_Disease</span></a>
<a class="sourceLine" id="cb6-5" data-line-number="5"><span class="co">#&gt;    &lt;chr&gt;     &lt;chr&gt;     &lt;chr&gt;       &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt;          </span></a>
<a class="sourceLine" id="cb6-6" data-line-number="6"><span class="co">#&gt;  1 ACH-0000… NIHOVCAR… NIH:OV…    905933      2201 Ovarian Cancer </span></a>
<a class="sourceLine" id="cb6-7" data-line-number="7"><span class="co">#&gt;  2 ACH-0000… HL60_HAE… HL-60      905938        55 Leukemia       </span></a>
<a class="sourceLine" id="cb6-8" data-line-number="8"><span class="co">#&gt;  3 ACH-0000… CACO2_LA… CACO2;…        NA        NA Colon/Colorect…</span></a>
<a class="sourceLine" id="cb6-9" data-line-number="9"><span class="co">#&gt;  4 ACH-0000… HEL_HAEM… HEL        907053       783 Leukemia       </span></a>
<a class="sourceLine" id="cb6-10" data-line-number="10"><span class="co">#&gt;  5 ACH-0000… HEL9217_… HEL 92…        NA        NA Leukemia       </span></a>
<a class="sourceLine" id="cb6-11" data-line-number="11"><span class="co">#&gt;  6 ACH-0000… MONOMAC6… MONO-M…    908148      2167 Leukemia       </span></a>
<a class="sourceLine" id="cb6-12" data-line-number="12"><span class="co">#&gt;  7 ACH-0000… LS513_LA… LS513      907795       569 Colon/Colorect…</span></a>
<a class="sourceLine" id="cb6-13" data-line-number="13"><span class="co">#&gt;  8 ACH-0000… C2BBE1_L… C2BBe1     910700      2104 Colon/Colorect…</span></a>
<a class="sourceLine" id="cb6-14" data-line-number="14"><span class="co">#&gt;  9 ACH-0000… NCIH2077… NCI-H2…        NA        NA Lung Cancer    </span></a>
<a class="sourceLine" id="cb6-15" data-line-number="15"><span class="co">#&gt; 10 ACH-0000… 253J_URI… 253J           NA        NA Bladder Cancer </span></a>
<a class="sourceLine" id="cb6-16" data-line-number="16"><span class="co">#&gt; # … with 1,666 more rows, and 3 more variables: Subtype_Disease &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb6-17" data-line-number="17"><span class="co">#&gt; #   Gender &lt;chr&gt;, Source &lt;chr&gt;</span></a></code></pre></div>
<ol>
<li><strong>DepMap_ID</strong> - ID for Dependency Map project</li>
<li><strong>CCLE_Name</strong> - name from the <a href="https://portals.broadinstitute.org/ccle">Cancer Cell Line Encyclopedia (CCLE)</a></li>
<li><strong>Aliases</strong> - other names</li>
<li><strong>COSMIC_ID</strong> - <a href="https://cancer.sanger.ac.uk/cosmic">COSMIC</a> ID</li>
<li><strong>Sanger_ID</strong> - Sanger ID</li>
<li><strong>Primary_Disease</strong> - general disease of the cell line</li>
<li><strong>Subtype_Disease</strong> - more specific disease of the cell line</li>
<li><strong>Gender</strong> - sex (if known) of the patient</li>
<li><strong>Source</strong> - source of the cell line</li>
</ol>
<h3 id="cell_line_mutationstib">cell_line_mutations.tib</h3>
<p>The mutation data for all cell lines in the Broad’s <a href="https://portals.broadinstitute.org/ccle">Cancer Cell Line Encyclopedia (CCLE)</a>.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">cell_line_mutations &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">file.path</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;cell_line_mutations.tib&quot;</span>))</a>
<a class="sourceLine" id="cb7-2" data-line-number="2">cell_line_mutations</a>
<a class="sourceLine" id="cb7-3" data-line-number="3"><span class="co">#&gt; # A tibble: 1,243,145 x 35</span></a>
<a class="sourceLine" id="cb7-4" data-line-number="4"><span class="co">#&gt;    Hugo_Symbol Entrez_Gene_Id NCBI_Build Chromosome Start_position</span></a>
<a class="sourceLine" id="cb7-5" data-line-number="5"><span class="co">#&gt;    &lt;chr&gt;                &lt;dbl&gt;      &lt;dbl&gt; &lt;chr&gt;               &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb7-6" data-line-number="6"><span class="co">#&gt;  1 VPS13D               55187         37 1                12359347</span></a>
<a class="sourceLine" id="cb7-7" data-line-number="7"><span class="co">#&gt;  2 AADACL4             343066         37 1                12726308</span></a>
<a class="sourceLine" id="cb7-8" data-line-number="8"><span class="co">#&gt;  3 IFNLR1              163702         37 1                24484172</span></a>
<a class="sourceLine" id="cb7-9" data-line-number="9"><span class="co">#&gt;  4 TMEM57               55219         37 1                25785018</span></a>
<a class="sourceLine" id="cb7-10" data-line-number="10"><span class="co">#&gt;  5 ZSCAN20               7579         37 1                33954141</span></a>
<a class="sourceLine" id="cb7-11" data-line-number="11"><span class="co">#&gt;  6 POU3F1                5453         37 1                38512139</span></a>
<a class="sourceLine" id="cb7-12" data-line-number="12"><span class="co">#&gt;  7 MAST2                23139         37 1                46498028</span></a>
<a class="sourceLine" id="cb7-13" data-line-number="13"><span class="co">#&gt;  8 GBP4                115361         37 1                89657103</span></a>
<a class="sourceLine" id="cb7-14" data-line-number="14"><span class="co">#&gt;  9 VAV3                 10451         37 1               108247170</span></a>
<a class="sourceLine" id="cb7-15" data-line-number="15"><span class="co">#&gt; 10 NBPF20           100288142         37 1               148346689</span></a>
<a class="sourceLine" id="cb7-16" data-line-number="16"><span class="co">#&gt; # … with 1,243,135 more rows, and 30 more variables: End_position &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb7-17" data-line-number="17"><span class="co">#&gt; #   Strand &lt;chr&gt;, Variant_Classification &lt;chr&gt;, Variant_Type &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb7-18" data-line-number="18"><span class="co">#&gt; #   Reference_Allele &lt;chr&gt;, Tumor_Seq_Allele1 &lt;chr&gt;, dbSNP_RS &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb7-19" data-line-number="19"><span class="co">#&gt; #   dbSNP_Val_Status &lt;chr&gt;, Genome_Change &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb7-20" data-line-number="20"><span class="co">#&gt; #   Annotation_Transcript &lt;chr&gt;, Tumor_Sample_Barcode &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb7-21" data-line-number="21"><span class="co">#&gt; #   cDNA_Change &lt;chr&gt;, Codon_Change &lt;chr&gt;, Protein_Change &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb7-22" data-line-number="22"><span class="co">#&gt; #   isDeleterious &lt;lgl&gt;, isTCGAhotspot &lt;lgl&gt;, TCGAhsCnt &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb7-23" data-line-number="23"><span class="co">#&gt; #   isCOSMIChotspot &lt;lgl&gt;, COSMIChsCnt &lt;dbl&gt;, ExAC_AF &lt;dbl&gt;,</span></a>
<a class="sourceLine" id="cb7-24" data-line-number="24"><span class="co">#&gt; #   VA_WES_AC &lt;chr&gt;, CGA_WES_AC &lt;chr&gt;, SangerWES_AC &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb7-25" data-line-number="25"><span class="co">#&gt; #   SangerRecalibWES_AC &lt;chr&gt;, RNAseq_AC &lt;chr&gt;, HC_AC &lt;chr&gt;, RD_AC &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb7-26" data-line-number="26"><span class="co">#&gt; #   WGS_AC &lt;chr&gt;, Variant_annotation &lt;chr&gt;, DepMap_ID &lt;chr&gt;</span></a></code></pre></div>
<ol>
<li><p><strong>Hugo_Symbol</strong> - gene name with the mutation</p></li>
<li><p><strong>Entrez_Gene_Id</strong> - Entrez ID</p></li>
<li><p><strong>NCBI_Build</strong> - reference genome used for mutation calling</p></li>
<li><p><strong>Chromosome</strong> - chromosome of the mutation</p></li>
<li><p><strong>Start_position</strong> - start position of the mutation</p></li>
<li><p><strong>End_position</strong> - end position of the mutation</p></li>
<li><p><strong>Strand</strong> - which strand the gene is on (transcribed from)</p></li>
<li><p><strong>Variant_Classification</strong> - the consequence of the mutation</p></li>
<li><p><strong>Variant_Type</strong> - shorthand consequence of the mutation (DEL, DNP, INS, ONP, SNP, or TN)</p></li>
<li><p><strong>Reference_Allele</strong> - nucleotide(s) in the reference</p></li>
<li><p><strong>Tumor_Seq_Allele1</strong> - nucleotide(s) in the sample</p></li>
<li><p><strong>dbSNP_RS</strong>, <strong>dbSNP_Val_Status</strong> - the ID association in the dbSNP (if available)</p></li>
<li><p><strong>Genome_Change</strong> - change to the genome</p></li>
<li><p><strong>Annotation_Transcript</strong> - transcript modified by the mutation</p></li>
<li><p><strong>Tumor_Sample_Barcode</strong> - name of the sample (cell line)</p></li>
<li><p><strong>cDNA_Change</strong> - change to the cDNA</p></li>
<li><p><strong>Codon_Change</strong> - the reference and mutated codon</p></li>
<li><p><strong>Protein_Change</strong> - amino acid change</p></li>
<li><p><strong>isDeleterious</strong>, <strong>Variant_annotation</strong>, <strong>isTCGAhotspot</strong>, <strong>TCGAhsCnt</strong>, <strong>isCOSMIChotspot</strong>, <strong>COSMIChsCnt</strong> - functional annotation</p></li>
<li><p><strong>ExAC_AF</strong>, <strong>VA_WES_AC</strong>, <strong>CGA_WES_AC</strong>, <strong>SangerWES_AC</strong>, <strong>SangerRecalibWES_AC</strong>, <strong>RNAseq_AC</strong>, <strong>HC_AC</strong>, <strong>RD_AC</strong>, <strong>WGS_AC</strong> - various accession IDs</p></li>
<li><p><strong>DepMap_ID</strong> - cell line ID for DepMap</p></li>
</ol>
<h3 id="cell_line_ras_annotib">cell_line_ras_anno.tib</h3>
<p>The cell line information with the status of the <em>RAS</em> isoforms. If they are all wild-type, then the <code>ras</code> and <code>allele</code> columns will both be <code>&quot;WT&quot;</code>. If a cell line has multiple RAS mutations, each one is a separate row.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">cell_line_ras_anno &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">file.path</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;cell_line_ras_anno.tib&quot;</span>))</a>
<a class="sourceLine" id="cb8-2" data-line-number="2">cell_line_ras_anno</a>
<a class="sourceLine" id="cb8-3" data-line-number="3"><span class="co">#&gt; # A tibble: 1,745 x 12</span></a>
<a class="sourceLine" id="cb8-4" data-line-number="4"><span class="co">#&gt;    DepMap_ID CCLE_Name Aliases COSMIC_ID Sanger_ID Primary_Disease</span></a>
<a class="sourceLine" id="cb8-5" data-line-number="5"><span class="co">#&gt;    &lt;chr&gt;     &lt;chr&gt;     &lt;chr&gt;       &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt;          </span></a>
<a class="sourceLine" id="cb8-6" data-line-number="6"><span class="co">#&gt;  1 ACH-0000… NIHOVCAR… NIH:OV…    905933      2201 Ovarian Cancer </span></a>
<a class="sourceLine" id="cb8-7" data-line-number="7"><span class="co">#&gt;  2 ACH-0000… HL60_HAE… HL-60      905938        55 Leukemia       </span></a>
<a class="sourceLine" id="cb8-8" data-line-number="8"><span class="co">#&gt;  3 ACH-0000… CACO2_LA… CACO2;…        NA        NA Colon/Colorect…</span></a>
<a class="sourceLine" id="cb8-9" data-line-number="9"><span class="co">#&gt;  4 ACH-0000… HEL_HAEM… HEL        907053       783 Leukemia       </span></a>
<a class="sourceLine" id="cb8-10" data-line-number="10"><span class="co">#&gt;  5 ACH-0000… HEL9217_… HEL 92…        NA        NA Leukemia       </span></a>
<a class="sourceLine" id="cb8-11" data-line-number="11"><span class="co">#&gt;  6 ACH-0000… MONOMAC6… MONO-M…    908148      2167 Leukemia       </span></a>
<a class="sourceLine" id="cb8-12" data-line-number="12"><span class="co">#&gt;  7 ACH-0000… LS513_LA… LS513      907795       569 Colon/Colorect…</span></a>
<a class="sourceLine" id="cb8-13" data-line-number="13"><span class="co">#&gt;  8 ACH-0000… C2BBE1_L… C2BBe1     910700      2104 Colon/Colorect…</span></a>
<a class="sourceLine" id="cb8-14" data-line-number="14"><span class="co">#&gt;  9 ACH-0000… NCIH2077… NCI-H2…        NA        NA Lung Cancer    </span></a>
<a class="sourceLine" id="cb8-15" data-line-number="15"><span class="co">#&gt; 10 ACH-0000… 253J_URI… 253J           NA        NA Bladder Cancer </span></a>
<a class="sourceLine" id="cb8-16" data-line-number="16"><span class="co">#&gt; # … with 1,735 more rows, and 6 more variables: Subtype_Disease &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb8-17" data-line-number="17"><span class="co">#&gt; #   Gender &lt;chr&gt;, Source &lt;chr&gt;, ras &lt;chr&gt;, allele &lt;chr&gt;, ras_allele &lt;chr&gt;</span></a></code></pre></div>
<ol>
<li><strong>DepMap_ID</strong> through 9. <strong>Source</strong> - same as for “cell_line_metadata.tib”</li>
<li><strong>ras</strong> - <em>RAS</em> isoform; <code>&quot;WT&quot;</code> means all are wild-type</li>
<li><strong>allele</strong> - mutant <em>RAS</em> allele; <code>&quot;WT&quot;</code> means all <em>RAS</em> isoforms are wild-type</li>
<li><strong>ras_allele</strong> - catenation of <code>ras</code> and <code>allele</code> columns</li>
</ol>
<h3 id="ras_muts_annotatedtib">ras_muts_annotated.tib</h3>
<p>The <em>RAS</em> mutant cell lines. This if the same data as in “cell_line_ras_anno.tib”, but only <em>RAS</em> mutants. The columns are in a different order, but hold the same data.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1">ras_muts_annotated &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">file.path</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;ras_muts_annotated.tib&quot;</span>))</a>
<a class="sourceLine" id="cb9-2" data-line-number="2">ras_muts_annotated</a>
<a class="sourceLine" id="cb9-3" data-line-number="3"><span class="co">#&gt; # A tibble: 420 x 12</span></a>
<a class="sourceLine" id="cb9-4" data-line-number="4"><span class="co">#&gt;    DepMap_ID ras   allele ras_allele CCLE_Name Aliases COSMIC_ID Sanger_ID</span></a>
<a class="sourceLine" id="cb9-5" data-line-number="5"><span class="co">#&gt;    &lt;chr&gt;     &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;      &lt;chr&gt;     &lt;chr&gt;       &lt;dbl&gt;     &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb9-6" data-line-number="6"><span class="co">#&gt;  1 ACH-0000… NRAS  Q61L   NRAS_Q61L  HL60_HAE… HL-60      905938        55</span></a>
<a class="sourceLine" id="cb9-7" data-line-number="7"><span class="co">#&gt;  2 ACH-0000… KRAS  G12D   KRAS_G12D  LS513_LA… LS513      907795       569</span></a>
<a class="sourceLine" id="cb9-8" data-line-number="8"><span class="co">#&gt;  3 ACH-0000… HRAS  G12V   HRAS_G12V  T24_URIN… T24        724812      1455</span></a>
<a class="sourceLine" id="cb9-9" data-line-number="9"><span class="co">#&gt;  4 ACH-0000… KRAS  G12V   KRAS_G12V  PATU8988… PA-TU-…        NA        NA</span></a>
<a class="sourceLine" id="cb9-10" data-line-number="10"><span class="co">#&gt;  5 ACH-0000… KRAS  G12V   KRAS_G12V  PATU8988… PA-TU-…   1240201      1242</span></a>
<a class="sourceLine" id="cb9-11" data-line-number="11"><span class="co">#&gt;  6 ACH-0000… NRAS  Q61K   NRAS_Q61K  CH157MN_… CH-157…        NA        NA</span></a>
<a class="sourceLine" id="cb9-12" data-line-number="12"><span class="co">#&gt;  7 ACH-0000… KRAS  Q61R   KRAS_Q61R  PANC0213… Panc 0…        NA        NA</span></a>
<a class="sourceLine" id="cb9-13" data-line-number="13"><span class="co">#&gt;  8 ACH-0000… NRAS  Q61L   NRAS_Q61L  PLB985_H… PLB985…        NA        NA</span></a>
<a class="sourceLine" id="cb9-14" data-line-number="14"><span class="co">#&gt;  9 ACH-0000… KRAS  G12D   KRAS_G12D  PANC0203… Panc 0…   1298475      1838</span></a>
<a class="sourceLine" id="cb9-15" data-line-number="15"><span class="co">#&gt; 10 ACH-0000… NRAS  G13D   NRAS_G13D  NCIH929_… NCI-H9…    724825      1230</span></a>
<a class="sourceLine" id="cb9-16" data-line-number="16"><span class="co">#&gt; # … with 410 more rows, and 4 more variables: Primary_Disease &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb9-17" data-line-number="17"><span class="co">#&gt; #   Subtype_Disease &lt;chr&gt;, Gender &lt;chr&gt;, Source &lt;chr&gt;</span></a></code></pre></div>
<ol>
<li><strong>DepMap_ID</strong> - cell line ID for DepMap</li>
<li><strong>ras</strong> - <em>RAS</em> isoform; <code>&quot;WT&quot;</code> means all are wild-type</li>
<li><strong>allele</strong> - mutant <em>RAS</em> allele; <code>&quot;WT&quot;</code> means all <em>RAS</em> isoforms are wild-type</li>
<li><strong>ras_allele</strong> - catenation of <code>ras</code> and <code>allele</code> columns</li>
<li><strong>CCLE_Name</strong> - name from the <a href="https://portals.broadinstitute.org/ccle">Cancer Cell Line Encyclopedia (CCLE)</a></li>
<li><strong>Aliases</strong> - other names</li>
<li><strong>COSMIC_ID</strong> - <a href="https://cancer.sanger.ac.uk/cosmic">COSMIC</a> ID</li>
<li><strong>Sanger_ID</strong> - Sanger ID</li>
<li><strong>Primary_Disease</strong> - general disease of the cell line</li>
<li><strong>Subtype_Disease</strong> - more specific disease of the cell line</li>
<li><strong>Gender</strong> - sex (if known) of the patient</li>
<li><strong>Source</strong> - source of the cell line</li>
</ol>
<h3 id="synthetic_lethal">synthetic_lethal</h3>
<p>I had to split up the synthetic lethal data by tissue so that each data file was small enough to push to GitHub. These are stored in “data/synthetic_lethal/”. All or a selection of them can be loaded using <code>load_all_synthetic_lethal</code>. It returns a single tibble of the desired data.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1">load_all_synthetic_lethal &lt;-<span class="st"> </span><span class="cf">function</span>(<span class="dt">tissues =</span> <span class="st">&quot;all&quot;</span>) {</a>
<a class="sourceLine" id="cb10-2" data-line-number="2">    tissues &lt;-<span class="st"> </span><span class="kw">paste0</span>(tissues, <span class="dt">collapse =</span> <span class="st">&quot;|&quot;</span>)</a>
<a class="sourceLine" id="cb10-3" data-line-number="3">    synlet_path &lt;-<span class="st"> </span><span class="kw">file.path</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;synthetic_lethal&quot;</span>)</a>
<a class="sourceLine" id="cb10-4" data-line-number="4">    synlet_files &lt;-<span class="st"> </span><span class="kw">list.files</span>(synlet_path, <span class="dt">full.name =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb10-5" data-line-number="5">                               <span class="dt">pattern =</span> <span class="st">&quot;_syn_lethal.tib&quot;</span>)</a>
<a class="sourceLine" id="cb10-6" data-line-number="6">    <span class="cf">if</span> (tissues <span class="op">!=</span><span class="st"> &quot;all&quot;</span>) {</a>
<a class="sourceLine" id="cb10-7" data-line-number="7">        synlet_files &lt;-<span class="st">  </span>stringr<span class="op">::</span><span class="kw">str_subset</span>(synlet_files, tissues)</a>
<a class="sourceLine" id="cb10-8" data-line-number="8">    }</a>
<a class="sourceLine" id="cb10-9" data-line-number="9">    synlet &lt;-<span class="st"> </span>purrr<span class="op">::</span><span class="kw">map</span>(synlet_files, readRDS) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">bind_rows</span>()</a>
<a class="sourceLine" id="cb10-10" data-line-number="10">    <span class="kw">return</span>(synlet)</a>
<a class="sourceLine" id="cb10-11" data-line-number="11">}</a></code></pre></div>
<p>A specific selection of tissues can be loaded by passing a vector of the tissue names (from the file names).</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">cervix_synlet &lt;-<span class="st"> </span><span class="kw">load_all_synthetic_lethal</span>(<span class="kw">c</span>(<span class="st">&quot;CERVIX&quot;</span>, <span class="st">&quot;BONE&quot;</span>))</a>
<a class="sourceLine" id="cb11-2" data-line-number="2">cervix_synlet</a>
<a class="sourceLine" id="cb11-3" data-line-number="3"><span class="co">#&gt; # A tibble: 328,871 x 15</span></a>
<a class="sourceLine" id="cb11-4" data-line-number="4"><span class="co">#&gt;    gene  cell_line    score DepMap_ID Aliases COSMIC_ID Sanger_ID</span></a>
<a class="sourceLine" id="cb11-5" data-line-number="5"><span class="co">#&gt;    &lt;chr&gt; &lt;chr&gt;        &lt;dbl&gt; &lt;chr&gt;     &lt;chr&gt;       &lt;dbl&gt;     &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb11-6" data-line-number="6"><span class="co">#&gt;  1 A1BG  143B_BONE  0.146   ACH-0010… &lt;NA&gt;           NA        NA</span></a>
<a class="sourceLine" id="cb11-7" data-line-number="7"><span class="co">#&gt;  2 NAT2  143B_BONE  0.103   ACH-0010… &lt;NA&gt;           NA        NA</span></a>
<a class="sourceLine" id="cb11-8" data-line-number="8"><span class="co">#&gt;  3 ADA   143B_BONE  0.169   ACH-0010… &lt;NA&gt;           NA        NA</span></a>
<a class="sourceLine" id="cb11-9" data-line-number="9"><span class="co">#&gt;  4 CDH2  143B_BONE  0.0630  ACH-0010… &lt;NA&gt;           NA        NA</span></a>
<a class="sourceLine" id="cb11-10" data-line-number="10"><span class="co">#&gt;  5 AKT3  143B_BONE -0.00808 ACH-0010… &lt;NA&gt;           NA        NA</span></a>
<a class="sourceLine" id="cb11-11" data-line-number="11"><span class="co">#&gt;  6 MED6  143B_BONE -0.214   ACH-0010… &lt;NA&gt;           NA        NA</span></a>
<a class="sourceLine" id="cb11-12" data-line-number="12"><span class="co">#&gt;  7 NR2E3 143B_BONE -0.154   ACH-0010… &lt;NA&gt;           NA        NA</span></a>
<a class="sourceLine" id="cb11-13" data-line-number="13"><span class="co">#&gt;  8 NAAL… 143B_BONE  0.134   ACH-0010… &lt;NA&gt;           NA        NA</span></a>
<a class="sourceLine" id="cb11-14" data-line-number="14"><span class="co">#&gt;  9 DUXB  143B_BONE  0.139   ACH-0010… &lt;NA&gt;           NA        NA</span></a>
<a class="sourceLine" id="cb11-15" data-line-number="15"><span class="co">#&gt; 10 PDZK… 143B_BONE  0.0303  ACH-0010… &lt;NA&gt;           NA        NA</span></a>
<a class="sourceLine" id="cb11-16" data-line-number="16"><span class="co">#&gt; # … with 328,861 more rows, and 8 more variables: Primary_Disease &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb11-17" data-line-number="17"><span class="co">#&gt; #   Subtype_Disease &lt;chr&gt;, Gender &lt;chr&gt;, Source &lt;chr&gt;, ras &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb11-18" data-line-number="18"><span class="co">#&gt; #   allele &lt;chr&gt;, ras_allele &lt;chr&gt;, tissue &lt;chr&gt;</span></a></code></pre></div>
<p>Or all tissues can be gathered by not passing anything.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1">synthetic_lethal &lt;-<span class="st"> </span><span class="kw">load_all_synthetic_lethal</span>()</a>
<a class="sourceLine" id="cb12-2" data-line-number="2">synthetic_lethal</a>
<a class="sourceLine" id="cb12-3" data-line-number="3"><span class="co">#&gt; # A tibble: 12,808,660 x 15</span></a>
<a class="sourceLine" id="cb12-4" data-line-number="4"><span class="co">#&gt;    gene  cell_line    score DepMap_ID Aliases COSMIC_ID Sanger_ID</span></a>
<a class="sourceLine" id="cb12-5" data-line-number="5"><span class="co">#&gt;    &lt;chr&gt; &lt;chr&gt;        &lt;dbl&gt; &lt;chr&gt;     &lt;chr&gt;       &lt;dbl&gt;     &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb12-6" data-line-number="6"><span class="co">#&gt;  1 A1BG  KELLY_AU…  NA      ACH-0002… KELLY      753618       225</span></a>
<a class="sourceLine" id="cb12-7" data-line-number="7"><span class="co">#&gt;  2 NAT2  KELLY_AU…  NA      ACH-0002… KELLY      753618       225</span></a>
<a class="sourceLine" id="cb12-8" data-line-number="8"><span class="co">#&gt;  3 ADA   KELLY_AU…  NA      ACH-0002… KELLY      753618       225</span></a>
<a class="sourceLine" id="cb12-9" data-line-number="9"><span class="co">#&gt;  4 CDH2  KELLY_AU…   0.133  ACH-0002… KELLY      753618       225</span></a>
<a class="sourceLine" id="cb12-10" data-line-number="10"><span class="co">#&gt;  5 AKT3  KELLY_AU…   0.142  ACH-0002… KELLY      753618       225</span></a>
<a class="sourceLine" id="cb12-11" data-line-number="11"><span class="co">#&gt;  6 MED6  KELLY_AU…  -0.410  ACH-0002… KELLY      753618       225</span></a>
<a class="sourceLine" id="cb12-12" data-line-number="12"><span class="co">#&gt;  7 NR2E3 KELLY_AU…  -0.0657 ACH-0002… KELLY      753618       225</span></a>
<a class="sourceLine" id="cb12-13" data-line-number="13"><span class="co">#&gt;  8 NAAL… KELLY_AU…  NA      ACH-0002… KELLY      753618       225</span></a>
<a class="sourceLine" id="cb12-14" data-line-number="14"><span class="co">#&gt;  9 DUXB  KELLY_AU…  NA      ACH-0002… KELLY      753618       225</span></a>
<a class="sourceLine" id="cb12-15" data-line-number="15"><span class="co">#&gt; 10 PDZK… KELLY_AU…  NA      ACH-0002… KELLY      753618       225</span></a>
<a class="sourceLine" id="cb12-16" data-line-number="16"><span class="co">#&gt; # … with 12,808,650 more rows, and 8 more variables:</span></a>
<a class="sourceLine" id="cb12-17" data-line-number="17"><span class="co">#&gt; #   Primary_Disease &lt;chr&gt;, Subtype_Disease &lt;chr&gt;, Gender &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb12-18" data-line-number="18"><span class="co">#&gt; #   Source &lt;chr&gt;, ras &lt;chr&gt;, allele &lt;chr&gt;, ras_allele &lt;chr&gt;, tissue &lt;chr&gt;</span></a></code></pre></div>
<ol>
<li><strong>gene</strong> -</li>
<li><strong>cell_line</strong> -</li>
<li><strong>score</strong> -</li>
<li><strong>DepMap_ID</strong> - cell line ID for DepMap</li>
<li><strong>Aliases</strong> - other names</li>
<li><strong>COSMIC_ID</strong> - <a href="https://cancer.sanger.ac.uk/cosmic">COSMIC</a> ID</li>
<li><strong>Sanger_ID</strong> - Sanger ID</li>
<li><strong>Primary_Disease</strong> - general disease of the cell line</li>
<li><strong>Subtype_Disease</strong> - more specific disease of the cell line</li>
<li><strong>Gender</strong> - sex (if known) of the patient</li>
<li><strong>Source</strong> - source of the cell line</li>
<li><strong>ras</strong> - <em>RAS</em> isoform; <code>&quot;WT&quot;</code> means all are wild-type</li>
<li><strong>allele</strong> - mutant <em>RAS</em> allele; <code>&quot;WT&quot;</code> means all <em>RAS</em> isoforms are wild-type</li>
<li><strong>ras_allele</strong> - catenation of <code>ras</code> and <code>allele</code> columns</li>
<li><strong>tissue</strong> - tissue of origin</li>
</ol>
<hr />
<h2 id="graphs">Graphs</h2>
<h3 id="ras_dependency_graphgr">ras_dependency_graph.gr</h3>
<p>A graph was created for the dependency score in each <em>RAS</em>-mutant cell line. Each edge represents a score between a <em>RAS</em> allele and gene (including the <em>RAS</em> isoforms).</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="kw">library</span>(tidygraph)</a>
<a class="sourceLine" id="cb13-2" data-line-number="2">ras_dependency_graph &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">file.path</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;ras_dependency_graph.gr&quot;</span>))</a>
<a class="sourceLine" id="cb13-3" data-line-number="3">ras_dependency_graph</a>
<a class="sourceLine" id="cb13-4" data-line-number="4"><span class="co">#&gt; # A tbl_graph: 17325 nodes and 1938496 edges</span></a>
<a class="sourceLine" id="cb13-5" data-line-number="5"><span class="co">#&gt; #</span></a>
<a class="sourceLine" id="cb13-6" data-line-number="6"><span class="co">#&gt; # An undirected multigraph with 1 component</span></a>
<a class="sourceLine" id="cb13-7" data-line-number="7"><span class="co">#&gt; #</span></a>
<a class="sourceLine" id="cb13-8" data-line-number="8"><span class="co">#&gt; # Node Data: 17,325 x 2 (active)</span></a>
<a class="sourceLine" id="cb13-9" data-line-number="9"><span class="co">#&gt;   name       gene_group</span></a>
<a class="sourceLine" id="cb13-10" data-line-number="10"><span class="co">#&gt;   &lt;chr&gt;      &lt;chr&gt;     </span></a>
<a class="sourceLine" id="cb13-11" data-line-number="11"><span class="co">#&gt; 1 KRAS_G12D  ras       </span></a>
<a class="sourceLine" id="cb13-12" data-line-number="12"><span class="co">#&gt; 2 KRAS_A146T ras       </span></a>
<a class="sourceLine" id="cb13-13" data-line-number="13"><span class="co">#&gt; 3 KRAS_G12C  ras       </span></a>
<a class="sourceLine" id="cb13-14" data-line-number="14"><span class="co">#&gt; 4 KRAS_G12V  ras       </span></a>
<a class="sourceLine" id="cb13-15" data-line-number="15"><span class="co">#&gt; 5 HRAS_Q61L  ras       </span></a>
<a class="sourceLine" id="cb13-16" data-line-number="16"><span class="co">#&gt; 6 KRAS_Q61H  ras       </span></a>
<a class="sourceLine" id="cb13-17" data-line-number="17"><span class="co">#&gt; # … with 1.732e+04 more rows</span></a>
<a class="sourceLine" id="cb13-18" data-line-number="18"><span class="co">#&gt; #</span></a>
<a class="sourceLine" id="cb13-19" data-line-number="19"><span class="co">#&gt; # Edge Data: 1,938,496 x 10</span></a>
<a class="sourceLine" id="cb13-20" data-line-number="20"><span class="co">#&gt;    from    to cell_line tissue  score ras   allele codon Primary_Disease</span></a>
<a class="sourceLine" id="cb13-21" data-line-number="21"><span class="co">#&gt;   &lt;int&gt; &lt;int&gt; &lt;chr&gt;     &lt;chr&gt;   &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt; &lt;chr&gt;          </span></a>
<a class="sourceLine" id="cb13-22" data-line-number="22"><span class="co">#&gt; 1     1    17 A427_LUNG LUNG   0.0837 KRAS  G12D   12    Lung Cancer    </span></a>
<a class="sourceLine" id="cb13-23" data-line-number="23"><span class="co">#&gt; 2     1    18 A427_LUNG LUNG   0.0458 KRAS  G12D   12    Lung Cancer    </span></a>
<a class="sourceLine" id="cb13-24" data-line-number="24"><span class="co">#&gt; 3     1    19 A427_LUNG LUNG   0.101  KRAS  G12D   12    Lung Cancer    </span></a>
<a class="sourceLine" id="cb13-25" data-line-number="25"><span class="co">#&gt; # … with 1.938e+06 more rows, and 1 more variable: Subtype_Disease &lt;chr&gt;</span></a></code></pre></div>
<p>Node attributes:</p>
<ol>
<li><strong>name</strong> - name of the node; either a <code>ras_allele</code> or gene name</li>
<li><strong>gene_group</strong> - either <code>&quot;ras&quot;</code> or <code>&quot;target&quot;</code></li>
</ol>
<p>Edge attributes:</p>
<ol>
<li><strong>from</strong>, <strong>to</strong> - source and destination of the edge (undirected graph)</li>
<li><strong>cell_line</strong> - cell line the score is from</li>
<li><strong>tissue</strong> - source tissue of the cell line</li>
<li><strong>score</strong> - dependency score</li>
<li><strong>ras</strong> - <em>RAS</em> isoform of the data point</li>
<li><strong>allele</strong> - <em>RAS</em> allele of the data point</li>
<li><strong>codon</strong> - which codon the allele is from (is a character, not numeric)</li>
<li><strong>Primary_Disease</strong> - general disease of the cell line</li>
<li><strong>Subtype_Disease</strong> - more specific disease of the cell line</li>
</ol>
<hr />
<p>If there are any mistake or places for explanation, please open an <a href="https://github.com/jhrcook/tidy_Achilles/issues">issue</a> or create a pull request if you want to address it yourself.</p>

</body>
</html>
