---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r setup, include = FALSE}
library(knitr)
knitr::opts_chunk$set(comment = "#>",
                      collapse = TRUE,
                      echo = TRUE)
```

# Tidy Project Achilles Data

**author: Joshua H. Cook**

**last updated: March 15, 2019**

This is "tidy" data from the Broad's [Project Achilles](https://depmap.org/portal/achilles/) with a focus of the various *RAS* alleles. The data files are in the "data" directory and exaplined below.

```{r load_libs}
library(tidyverse)
```

To download this repository, run the following command on the command line.

```bash
git clone https://github.com/jhrcook/tidy_Achilles.git
```

---

## Raw Data

### DepMap-2019q1-celllines_v2.csv

Information on cell lines from the Broad's [Cancer Cell Line Encyclopedia (CCLE)](https://portals.broadinstitute.org/ccle). The tidy data is available in "cell_line_metadata.tib".

### depmap_19Q1_mutation_calls.csv

Mutation data on the cell lines from the Broad's [Cancer Cell Line Encyclopedia (CCLE)](https://portals.broadinstitute.org/ccle). The tidy data is available in "cell_line_mutations.tib".

To access the file, decompress it with a GUI tools (usually double-click on Mac works) or use the command line.

```bash
gunzip depmap_19Q1_mutation_calls.csv.gz
```

### D2_combined_gene_dep_scores.csv

The "dependency scores" calculated by the [Achilles Project RNAi screen](https://depmap.org/portal/achilles/). This file is organized by target name in the first column and the following columns are the scores for each cell line. The tidy data is available in "synthetic_lethal.tib".

To access the file, decompress it with a GUI tools (usually double-click on Mac works) or use the command line.

```bash
gunzip D2_combined_gene_dep_scores.csv.gz
```

### To be added

* [DepMap gene dependency scores](https://depmap.org/portal/download/) (more recently updated) which uses the CRISPR screen results
* Genes  marked as "essential" and  "nonessential" by DepMap
* Any others? Open an [issue](https://github.com/jhrcook/tidy_Achilles/issues) with for any requested data to include.

---

## Tidy Data Tables

All processing was done in "data_preparation.R". The tidy data were stored as "tibbles" (`tbl_df`, instead of R's standard data.frame object) in RDS files. They can all be read directly into R.

```{r example_readRDS, eval = FALSE}
library(tibble)
readRDS("data/example_data_table.tib")
```

More information in the "tidy data" format can be found in [*R for Data Science - Tidy data*](https://r4ds.had.co.nz/tidy-data.html). 

### cell_line_metadata.tib

The information for each cell line from the Broad's [Cancer Cell Line Encyclopedia (CCLE)](https://portals.broadinstitute.org/ccle).

```{r show_celllinemeta}
cell_line_metadata <- readRDS(file.path("data", "cell_line_metadata.tib"))
cell_line_metadata
```

1. **DepMap_ID** - ID for Dependency Map project
2. **CCLE_Name** - name from the [Cancer Cell Line Encyclopedia (CCLE)](https://portals.broadinstitute.org/ccle)
3. **Aliases** - other names
4. **COSMIC_ID** - [COSMIC](https://cancer.sanger.ac.uk/cosmic) ID
5. **Sanger_ID** - Sanger ID
6. **Primary_Disease** - general disease of the cell line
7. **Subtype_Disease** - more specific disease of the cell line
8. **Gender** - sex (if known) of the patient
9. **Source** - source of the cell line

### cell_line_mutations.tib

The mutation data for all cell lines in the Broad's [Cancer Cell Line Encyclopedia (CCLE)](https://portals.broadinstitute.org/ccle).

```{r show_cellline_mutations.tib}
cell_line_mutations <- readRDS(file.path("data", "cell_line_mutations.tib"))
cell_line_mutations
```

1. **Hugo_Symbol** - gene name with the mutation
2. **Entrez_Gene_Id** - Entrez ID
3. **NCBI_Build** - reference genome used for mutation calling
4. **Chromosome** - chromosome of the mutation
5. **Start_position** - start position of the mutation
6. **End_position** - end position of the mutation
7. **Strand** - which strand the gene is on (transcribed from)
8. **Variant_Classification** - the consequence of the mutation
9. **Variant_Type** - shorthand consequence of the mutation (DEL, DNP, INS, ONP, SNP, or TN)
10. **Reference_Allele** - nucleotide(s) in the reference
11. **Tumor_Seq_Allele1** - nucleotide(s) in the sample
12. **dbSNP_RS**, **dbSNP_Val_Status** - the ID association in the dbSNP (if available)
13. **Genome_Change** - change to the genome
14. **Annotation_Transcript** - transcript modified by the mutation
15. **Tumor_Sample_Barcode** - name of the sample (cell line)
16. **cDNA_Change** - change to the cDNA
17. **Codon_Change** - the reference and mutated codon
18. **Protein_Change** - amino acid change
19. **isDeleterious**, **Variant_annotation**, **isTCGAhotspot**, **TCGAhsCnt**, **isCOSMIChotspot**, **COSMIChsCnt** - functional annotation

20. **ExAC_AF**, **VA_WES_AC**, **CGA_WES_AC**, **SangerWES_AC**, **SangerRecalibWES_AC**, **RNAseq_AC**, **HC_AC**, **RD_AC**, **WGS_AC** -  various accession IDs
21. **DepMap_ID** - cell line ID for DepMap


### cell_line_ras_anno.tib

The cell line information with the status of the *RAS* isoforms. If they are all wild-type, then the `ras` and `allele` columns will both be `"WT"`. If a cell line has multiple RAS mutations, each one is a separate row.

```{r show_celllinerasanno}
cell_line_ras_anno <- readRDS(file.path("data", "cell_line_ras_anno.tib"))
cell_line_ras_anno
```

1. **DepMap_ID** through **Source** - same as for "cell_line_metadata.tib"
2. **ras** - *RAS* isoform; `"WT"` means all are wild-type
3. **allele** - mutant *RAS* allele; `"WT"` means all *RAS* isoforms are wild-type
4. **ras_allele** - catenation of `ras` and  `allele` columns

### ras_muts_annotated.tib

The *RAS* mutant cell lines. This if the same data as in "cell_line_ras_anno.tib", but only *RAS* mutants. The columns are in  a different order, but hold the same data.

```{r show_rasmutsannotated}
ras_muts_annotated <- readRDS(file.path("data", "ras_muts_annotated.tib"))
ras_muts_annotated
```

1. **DepMap_ID** - cell line ID for DepMap
2. **ras** - *RAS* isoform; `"WT"` means all are wild-type
3. **allele** - mutant *RAS* allele; `"WT"` means all *RAS* isoforms are wild-type
4. **ras_allele** - catenation of `ras` and  `allele` columns
5. **CCLE_Name** - name from the [Cancer Cell Line Encyclopedia (CCLE)](https://portals.broadinstitute.org/ccle)
6. **Aliases** - other names
7. **COSMIC_ID** - [COSMIC](https://cancer.sanger.ac.uk/cosmic) ID
8. **Sanger_ID** - Sanger ID
9. **Primary_Disease** - general disease of the cell line
10. **Subtype_Disease** - more specific disease of the cell line
11. **Gender** - sex (if known) of the patient
12. **Source** - source of the cell line

Here is a chart to breakdown the number of cell lines for each *RAS* alleles across the tissues.

![ras_cell_lines](images/ras_alleles_per_tissue.png)

### synthetic_lethal

I had to split up the synthetic lethal data by tissue so that each data file was small enough to push to GitHub. These are stored in "data/synthetic_lethal/". All or a selection of them can be loaded using `load_all_synthetic_lethal`. It returns a single tibble of the desired data.

```{r load_snytheticlethal}
load_all_synthetic_lethal <- function(tissues = "all") {
    tissues <- paste0(tissues, collapse = "|")
    synlet_path <- file.path("data", "synthetic_lethal")
    synlet_files <- list.files(synlet_path, full.name = TRUE,
                               pattern = "_syn_lethal.tib")
    if (tissues != "all") {
        synlet_files <-  stringr::str_subset(synlet_files, tissues)
    }
    synlet <- purrr::map(synlet_files, readRDS) %>% bind_rows()
    return(synlet)
}
```

A specific selection of tissues can be loaded by passing a vector of the tissue names (from the file names).

```{r example_fewtissues}
cervix_synlet <- load_all_synthetic_lethal(c("CERVIX", "BONE"))
cervix_synlet
```

Or all tissues can be gathered by not passing anything.

```{r example_alltissues}
synthetic_lethal <- load_all_synthetic_lethal()
synthetic_lethal
```

1. **gene** - gene target
2. **cell_line** - cell line
3. **score** - dependency score (calculated by the Achilles project)
4. **DepMap_ID** - cell line ID for DepMap
5. **Aliases** - other names
6. **COSMIC_ID** - [COSMIC](https://cancer.sanger.ac.uk/cosmic) ID
7. **Sanger_ID** - Sanger ID
8. **Primary_Disease** - general disease of the cell line
9. **Subtype_Disease** - more specific disease of the cell line
10. **Gender** - sex (if known) of the patient
11. **Source** - source of the cell line
12. **ras** - *RAS* isoform; `"WT"` means all are wild-type
13. **allele** - mutant *RAS* allele; `"WT"` means all *RAS* isoforms are wild-type
14. **ras_allele** - catenation of `ras` and  `allele` columns
15. **tissue** - tissue of origin

---

## Graphs

### ras_dependency_graph.gr

A graph was created for the dependency score in each *RAS*-mutant cell line. Each edge represents a score between a *RAS* allele and gene (including the *RAS* isoforms).

```{r show_rasdependencygraph.gr}
library(tidygraph)
ras_dependency_graph <- readRDS(file.path("data", "ras_dependency_graph.gr"))
ras_dependency_graph
```

Node attributes:

1. **name** - name of the node; either a `ras_allele` or gene name
2. **gene_group** - either `"ras"` or `"target"`

Edge attributes:

1. **from**, **to** - source and destination of the edge (undirected graph)
2. **cell_line** - cell line the score is from
3. **tissue** - source tissue of the cell line
4. **score** - dependency score
5. **ras** - *RAS* isoform of the data point
6. **allele** - *RAS* allele of the data point
7. **codon** - which codon the allele is from (is a character, not numeric)
8. **Primary_Disease** - general disease of the cell line
9. **Subtype_Disease** - more specific disease of the cell line

Here is an example of the colorectal cancer cell lines with *KRAS* mutations in the hotspot codons, only showing edges for scores greater than 1 (red) or less than -1 (blue). If there were multiple scores for a target-*RAS* allele, the median value was used.

![dependency_map](images/dependency_graph_nicely.png)

Below are the dependency values each *RAS* allele on the corresponding *RAS* gene.

![ras_dependency](images/ras_dependence.png)

---

If there are any mistake or places for explanation, please open an [issue](https://github.com/jhrcook/tidy_Achilles/issues) or create a pull request if you want to address it yourself.
