---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r setup, include = FALSE}
library(knitr)
knitr::opts_chunk$set(comment = "#>",
                      collapse = TRUE,
                      echo = TRUE)
```

# Tidy Project Achilles Data

**author: Joshua H. Cook**

**last updated: March 15, 2019**

This is "tidy" data from the Broad's [Project Achilles](https://depmap.org/portal/achilles/) with a focus of the various *RAS* alleles. The data files are in the "data" directory and exaplined below.

```{r load_libs, message = FALSE, warching = FALSE}
library(tidyverse)
```

To download this repository, run the following command on the command line.

```bash
git clone https://github.com/jhrcook/tidy_Achilles.git
```

## Data Source

All data was downloaded from the [Broad's DepMap data repository](https://depmap.org/portal/download/all/).\
https://depmap.org/portal/download/all/

You can also query genes, cell lines, lineages, etc. from their [website](https://depmap.org/portal/)\
https://depmap.org/portal/

The cell line information was obtained from the [Cancer Cell Line Encyclopedia](https://portals.broadinstitute.org/ccle) - their query portal is really useful, too. More data on the cell lines can be downloaded from their website. If you have an requests for data to add to this repo, please open a GitHub [issue](https://github.com/jhrcook/tidy_Achilles/issues).\
https://portals.broadinstitute.org/ccle

---

## Raw Data

### DepMap-2019q1-celllines_v2.csv

Information on cell lines from the Broad's [Cancer Cell Line Encyclopedia (CCLE)](https://portals.broadinstitute.org/ccle). The tidy data is available in "cell\_line\_metadata.tib".

### depmap\_19Q1\_mutation\_calls.csv

Mutation data on the cell lines from the Broad's [Cancer Cell Line Encyclopedia (CCLE)](https://portals.broadinstitute.org/ccle). The tidy data is available in "cell\_line\_mutations.tib".

To access the file, decompress it with a GUI tool (usually double-click on Mac works) or use the command line.

```bash
gunzip data/depmap_19Q1_mutation_calls.csv.gz
```

### D2\_combined\_gene\_dep\_scores.csv

The "dependency scores" calculated by the [Achilles Project RNAi screen](https://depmap.org/portal/achilles/). This file is organized by target name in the first column and the following columns are the scores for each cell line. The tidy data is available in "synthetic_lethal.tib".

To access the file, decompress it with a GUI tool (usually double-click on Mac works) or use the command line.

```bash
gunzip data/D2_combined_gene_dep_scores.csv.gz
```

### nonessential\_genes.txt and essential\_genes.txt

Just a list of genes the DepMap project has declared as essential or nonessential in all cell lines. The tidy data is available in "gene_essentiality.tib".

### public\_19Q1\_gene\_cn.csv

A cell line x gene matrix of copy number calls (they use GATK). The data was broken down by cell line - see below for how to easily load the desired data.

### gene\_effect\_corrected.csv

**TODO**

```bash
gunzip data/gene_effect_corrected.csv.gz
```

### gene\_dependency\_corrected.csv

**TODO**

```bash
gunzip data/gene_dependency_corrected.csv.gz
```

### To be added

Any others? Open an [issue](https://github.com/jhrcook/tidy_Achilles/issues) for any requested data to include.


---

## Tidy Data Tables

All processing was done in "data_preparation.R". The tidy data were stored as "tibbles" (`tbl_df`, instead of R's standard data.frame object) in RDS files. They can all be read directly into R.

```{r example_readRDS, eval = FALSE}
library(tibble)
readRDS("data/example_data_table.tib")
```

More information in the "tidy data" format can be found in [*R for Data Science - Tidy data*](https://r4ds.had.co.nz/tidy-data.html). 


### cell\_line\_metadata.tib

The information for each cell line from the Broad's [Cancer Cell Line Encyclopedia (CCLE)](https://portals.broadinstitute.org/ccle).

```{r show_celllinemeta}
cell_line_metadata <- readRDS(file.path("data", "cell_line_metadata.tib"))
cell_line_metadata
```

**DepMap\_ID** - ID for Dependency Map project\
**CCLE\_Name** - name from the [Cancer Cell Line Encyclopedia (CCLE)](https://portals.broadinstitute.org/ccle)\
**Aliases** - other names\
**COSMIC\_ID** - [COSMIC](https://cancer.sanger.ac.uk/cosmic) ID\
**Sanger\_ID** - Sanger ID\
**Primary\_Disease** - general disease of the cell line\
**Subtype\_Disease** - more specific disease of the cell line\
**Gender** - sex (if known) of the patient\
**Source** - source of the cell line


### cell\_line\_mutations.tib

The mutation data for all cell lines in the Broad's [Cancer Cell Line Encyclopedia (CCLE)](https://portals.broadinstitute.org/ccle).

```{r show_cellline_mutations.tib}
cell_line_mutations <- readRDS(file.path("data", "cell_line_mutations.tib"))
cell_line_mutations
```

**Chromosome** - chromosome of the mutation\
**Start\_position** - start position of the mutation\
**End\_position** - end position of the mutation\
**Strand** - which strand the gene is on (transcribed from)\
**Variant\_Classification** - the consequence of the mutation\
**Variant\_Type** - shorthand consequence of the mutation (DEL, DNP, INS, ONP, SNP, or TN)\
**Reference\_Allele** - nucleotide(s) in the reference\
**Tumor\_Seq\_Allele1** - nucleotide(s) in the sample\
**dbSNP\_RS**, **dbSNP\_Val\_Status** - the ID association in the dbSNP (if available)\
**Genome\_Change** - change to the genome\
**Annotation\_Transcript** - transcript modified by the mutation\
**Tumor\_Sample\_Barcode** - name of the sample (cell line)\
**cDNA\_Change** - change to the cDNA\
**Codon\_Change** - the reference and mutated codon\
**Protein\_Change** - amino acid change\
**isDeleterious**, **Variant\_annotation**, **isTCGAhotspot**, **TCGAhsCnt**, **isCOSMIChotspot**, **COSMIChsCnt** - functional annotation\
**ExAC\_AF**, **VA\_WES\_AC**, **CGA\_WES\_AC**, **SangerWES\_AC**, **SangerRecalibWES\_AC**, **RNAseq\_AC**, **HC\_AC**, **RD\_AC**, **WGS\_AC** -  various accession IDs


### cell\_line\_ras\_anno.tib

The cell line information with the status of the *RAS* isoforms. If they are all wild-type, then the `ras` and `allele` columns will both be `"WT"`. If a cell line has multiple RAS mutations, each one is a separate row.

```{r show_celllinerasanno}
cell_line_ras_anno <- readRDS(file.path("data", "cell_line_ras_anno.tib"))
cell_line_ras_anno
```


### ras\_muts\_annotated.tib

The *RAS* mutant cell lines. This if the same data as in "cell\_line\_ras\_anno.tib", but only *RAS* mutants. The columns are in  a different order, but hold the same data.

```{r show_rasmutsannotated}
ras_muts_annotated <- readRDS(file.path("data", "ras_muts_annotated.tib"))
ras_muts_annotated
```

Here is a chart to breakdown the number of cell lines for each *RAS* alleles across the tissues.

![ras_cell_lines](images/ras_alleles_per_tissue.png)


### synthetic_lethal/

I had to split up the synthetic lethal data by tissue so that each data file was small enough to push to GitHub. These are stored in "data/synthetic_lethal/". All or a selection of them can be loaded using `load\_synthetic\_lethal`. It returns a single tibble of the desired data. The only new column here is `score` which holds the lethality score that DepMap calculated.

```{r load_snytheticlethal}
load_synthetic_lethal <- function(tissues = "all") {
    tissues <- paste0(tissues, collapse = "|")
    synlet_path <- file.path("data", "synthetic_lethal")
    synlet_files <- list.files(synlet_path, full.name = TRUE,
                               pattern = "_syn_lethal.tib")
    if (tissues != "all") {
        synlet_files <-  stringr::str_subset(synlet_files, tissues)
    }
    synlet <- purrr::map(synlet_files, readRDS) %>% bind_rows()
    return(synlet)
}
```

A specific selection of tissues can be loaded by passing a vector of the tissue names (from the file names).

```{r example_fewtissues}
cervix_synlet <- load_synthetic_lethal(c("CERVIX", "BONE"))
cervix_synlet
```

Or all tissues can be gathered by not passing anything.

```{r example_alltissues}
synthetic_lethal <- load_synthetic_lethal()
synthetic_lethal
```


### gene\_essentiality.tib

A two column tibble of what the DepMap project deems essential or not essential in all cell lines.

```{r sow_essential}
readRDS(file.path("data", "gene_essentiality.tib"))
```


### copy\_number/

The full tibble was too large to push to GitHub (and probably to warrant loading every single time), so I separated it by cell line and stored each as a tibble in "data/copy_number". Again, I supply a function below to retrieve each one.

```{r load_celllinecopynum}
load_copy_number <- function(cell_line = "all") {
    cell_line <- paste0(cell_line, collapse = "|")
    copynum_path <- file.path("data", "copy_number")
    copynum_files <- list.files(copynum_path, full.name = TRUE,
                               pattern = "_syn_lethal.tib")
    if (cell_line != "all") {
        copynum_files <-  stringr::str_subset(copynum_files, cell_line)
    }
    copynum <- purrr::map(copynum_files, readRDS) %>% bind_rows()
    return(copynum)
}
```

Here is an example.

```{r show_celllinecopynum}
load_copy_number("ACH-000690")
```

*I will add other cell line information in the future.*

---

## Graphs

### ras\_dependency\_graph.gr

A graph was created for the dependency score in each *RAS*-mutant cell line. Each edge represents a score between a *RAS* allele and gene (including the *RAS* isoforms).

```{r show_rasdependencygraph.gr}
library(tidygraph)
ras_dependency_graph <- readRDS(file.path("data", "ras_dependency_graph.gr"))
ras_dependency_graph
```

Here is an example of the colorectal cancer cell lines with *KRAS* mutations in the hotspot codons, only showing edges for scores greater than 1 (red) or less than -1 (blue). If there were multiple scores for a target-*RAS* allele, the median value was used.

![dependency_map](images/dependency_graph_nicely.png)

Below are the dependency values each *RAS* allele on the corresponding *RAS* gene.

![ras_dependency](images/ras_dependence.png)

---

If there are any mistake or places for explanation, please open an [issue](https://github.com/jhrcook/tidy_Achilles/issues) or create a pull request if you want to address it yourself.
